src/UuidGenerator.php

```diff
@@ -27,14 +27,32 @@ class UuidGenerator extends AbstractIdGenerator
     /**
      * Generate an identifier
      *
-     * @param EntityManager $em
+     * @param EntityManagerInterface $em
+     * @param Entity $entity
+     *
+     * @return UuidInterface
+     *
+     * @throws Exception
+     *
+     * @deprecated as per parent. Accepts any EntityManagerInterface for maximum compatibility on PHP 7.4+.
+     * @see UuidGenerator::generateId()
+     */
+    public function generate(EntityManagerInterface $em, $entity): UuidInterface
+    {
+        return Uuid::uuid4();
+    }
+
+    /**
+     * Generate an identifier
+     *
+     * @param EntityManagerInterface $em
      * @param Entity $entity
      *
      * @return UuidInterface
      *
      * @throws Exception
      */
-    public function generate(EntityManager $em, $entity)
+    public function generateId(EntityManagerInterface $em, $entity): UuidInterface
```

Comment:
This introduces a BC break, right? If anyone extends `UuidGenerator` and overrides this method, making this change will result in an error like:

> Fatal error: Declaration of Foo::generate(EntityManager $em, $entity) must be compatible with Ramsey\Uuid\Doctrine\UuidGenerator::generate(EntityManagerInterface $em, $entity): UuidInterface

Reply:
@ramsey Yes, sorry I wasn't clear. This is what I was trying to get at with:

> I suggest this targets a new major release.

I would also expect the bump in minimum PHP version to necessitate a new major version as that's also a BC break, hence my thinking that the two changes could go hand in hand.

Reply:
A bump in minimum PHP version shouldn't be considered a BC break IMHO. The reasoning is explained under "_Why dropping PHP support in a minor version is not a BC break_" at https://www.doctrine-project.org/2017/07/25/php-7.1-requirement-and-composer.html